src/backend/catalog/storage.c

```diff
@@ -115,7 +117,14 @@ RelationCreateStorage(RelFileNode rnode, char relpersistence, char relstorage)
 	smgrcreate(srel, MAIN_FORKNUM, false);
 
 	if (needs_wal)
+	{
+		/* 
+		 * Call GetCurrentTransactionId before log_smgrcreate, because
+		 * XLOG_SMGR_CREATE_PDL WAL record should be always linked to XID
+		 */
+		xid = GetCurrentTransactionId();
```

Comment:
May `GetCurrentTransactionId` return invalid value? If yes, should we check it here?

Reply:
It's written in the GetCurrentTransactionId function description: "This will return the XID of the current transaction (main or sub transaction), assigning one if it's not yet set". So the function should not return invalid value